sdk/monitor/Azure.Monitor.Ingestion/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,100 @@
+1. The permissions may take up to 30 minutes to propagate. If the permissions were granted recently, retry after some time.
+
+### Troubleshooting missing logs
+
+When you send logs to Azure Monitor for ingestion, the request may succeed, but you may not see the data appear in the designated Log Analytics workspace table as configured in the DCR. To investigate and resolve this issue, ensure the following:
+
+* Double-check that you're using the correct DCE when configuring the `LogsIngestionClient`. Using the wrong endpoint can result in data not being properly sent to the Log Analytics workspace.
+
+* Make sure you provide the correct DCR ID to the `Upload` method. The DCR ID is an immutable identifier that determines the transformation rules applied to the uploaded logs and directs them to the appropriate Log Analytics workspace table.
+
+* Verify that the custom table specified in the DCR exists in the Log Analytics workspace. Ensure that you provide the accurate name of the custom table to the upload method. Mismatched table names can lead to logs not being stored correctly.
+
+* Confirm that the logs you're sending adhere to the format expected by the DCR. The data should be in the form of a JSON object or array, structured according to the requirements specified in the DCR. Additionally, it's essential to encode the request body in UTF-8 to avoid any data transmission issues.
+
+* Keep in mind that data ingestion may take some time, especially if you're sending data to a specific table for the first time. In such cases, allow up to 15 minutes for the data to be fully ingested and available for querying and analysis.
+
+### Troubleshooting slow logs upload
+
+If you experience delays when uploading logs, it could be due to reaching service limits, which may trigger the rate limiter to throttle your client. To determine if your client has reached service limits, you can enable logging and check if the service is returning errors with an HTTP status code 429. For more information on service limits, see the [Azure Monitor service limits documentation](https://learn.microsoft.com/azure/azure-monitor/service-limits#logs-ingestion-api).
+
+To enable client logging and to troubleshoot this issue further, see the instructions provided in the section titled [Enable client logging](#enable-client-logging).
+
+If there are no throttling errors, then consider increasing the concurrency to upload multiple log requests in parallel.
+To set the concurrency, use the `UploadLogsOptions.MaxConcurrency` property.
+
+```C# Snippet:UploadWithMaxConcurrency
+var endpoint = new Uri("<data_collection_endpoint_uri>");
+var ruleId = "<data_collection_rule_id>";
+var streamName = "<stream_name>";
+
+var credential = new DefaultAzureCredential();
+LogsIngestionClient client = new(endpoint, credential);
```

Comment:
Let's be consistent with the object creation syntax in these code samples. Note the 2 different approaches taken in lines 79 and 80.